### DISCUSSION TITLE
Как сделать график если есть несколько параметров

### DISCUSSION THREAD

**Discussion by andlev7** [totumonline/totum-ru-issues-and-discussions#176] [Как это сделать?] [answered]
Делаю график. У меня таблица со следующими данными. 

![image](https://user-images.githubusercontent.com/52709157/182655633-e6254684-c37a-4cac-8ff8-ce9959f5a7d6.png)

Хочу сделать 3 графика в которых видно динамику по одной из метрик (visible, traffic, keywords)  по каждому домену по дате.
То есть нужно как-то передавать домены в шаблон JSON чтобы получилось вот так

![image](https://user-images.githubusercontent.com/52709157/182655912-22505298-b507-4009-a11d-2862f9c6b187.png)

Даже не знаю куда копать поскольку тут  получается что domain - параметр для разбивки, date- параметр, а Traffic - это показатель.

**Comment by ttm-alexeypolunin** [comment]
Полез разбираться и обнаружил, что у нас `label` для `datasets` прибит гвоздями железными. Сейчас думаем, что с этим делать.

**Reply by andlev7** [reply]
Спасибо, именно на этом я и застрял не понимая как передавать значения доменов в датасет

**Comment by ttm-alexeypolunin** [comment, marked as answer]
**Готово в версии 3.6.45.4**

### [Видео на YouTube](https://youtu.be/bNMPfTLfn9Q)

Файл примера: https://raw.githubusercontent.com/totumonline/ttm-samples/main/samples/totum_sample_13.json.gz.ttm

Инструкции по установке примеров: https://github.com/totumonline/ttm-samples

Ключевое изменение – динамические параметры `datasets` передаваемые в основном `row`: `field: "datasets_label" = $list_d_labels`

Ключ в `row` указывается как `datasets_PARAM`.

Код в графике:

```
=: rowCreate(field: "labels" = $list_labels; field: "values" = $list_values; field: "datasets_label" = $list_d_labels)

list_labels: $dates
list_d_labels: $sites
list_values: listReplace(list: $sites; action: $sites_data; value: "sites_data")
	sites_data: $create_data[[$column]]
		column: $#sites_data

~create_data: listReplace(list: $dates; action: $val; value: "date")
	val: rowCreateByLists(keys: $keysList; values: $valuesList)
		keysList: $sites
		valuesList: listReplace(list: $sites; action: $keywords; value: "site")
			keywords: select(table: $#ntn; field: 'keywords'; where: 'site' = $#site; where: 'date' = $#date)

~dates: listUniq(list: $list_dates)
		list_dates: selectList(table: $#ntn; field: 'date'; order: 'date' asc)

~sites: listUniq(list: $list_sites)
	list_sites: selectList(table: $#ntn; field: 'site'; order: 'site' asc)
```

**Reply by andlev7** [reply]
Спасибо большое! Отлично работает и масштабируемое решение

**Comment by andlev7** [comment]
Спасибо, именно на этом я и застрял не понимая как динамически добавлять
домены в лейбл

чт, 4 авг. 2022 г. в 16:05, Alexey Polunin ***@***.***>:
